libs/application-generic/src/usecases/workflow/update-workflow/update-workflow.usecase.ts

```diff
@@ -197,10 +184,106 @@ export class UpdateWorkflow {
       updatePayload.payloadSchema = command.payloadSchema;
     }
 
+    // defaultPreferences is required, so we always call the upsert
+    await this.upsertPreferences.upsertWorkflowPreferences(
+      UpsertWorkflowPreferencesCommand.create({
+        templateId: command.id,
+        preferences: command.defaultPreferences,
+        environmentId: command.environmentId,
+        organizationId: command.organizationId,
+      }),
+    );
+
+    if (
+      command.userPreferences !== undefined ||
+      command.critical !== undefined
+    ) {
+      /*
+       * userPreferences is optional, so we need to check if it's defined before calling the upsert.
+       * we also need to check if the legacy `critical` property is defined, because if provided,
+       * it's used to set the `userPreferences.all.readOnly` property
+       */
+
+      updatePayload.critical = command.critical;
+      this.analyticsService.track(
+        'Update Critical Template - [Platform]',
+        command.userId,
+        {
+          _organization: command.organizationId,
+          critical: command.userPreferences?.all?.readOnly,
+        },
+      );
+
+      /*
+       * This builder pattern is only needed for the `critical` property,
+       * ensuring it's set in the `userPreferences.all.readOnly` property
+       * when supplied.
+       *
+       * TODO: remove this once we deprecate the `critical` property
+       * and use only the `userPreferences` object
+       */
+      const defaultUserPreferences =
+        command.userPreferences ?? existingTemplate.userPreferences;
+      const defaultCritical =
+        command.userPreferences?.all?.readOnly ??
+        command.critical ??
+        existingTemplate.userPreferences?.all?.readOnly ??
+        existingTemplate.critical;
+
+      const userPreferences =
+        command.userPreferences === null
```

Comment:
Does this mean that user preferences are deleted?

Reply:
Yes. I added an explicit `DeletePreference` use-case now and we don't have this ternary anymore.